bin/conform-templates.php

```diff
@@ -0,0 +1,299 @@
+    fwrite(
+        STDERR,
+        'File composer.json does not exist, please call script from main repository directory'
+    );
+    exit(1);
+}
+
+$composer = json_decode(file_get_contents('composer.json'), true);
+if (! isset($composer['name']) || ! preg_match('/^([a-z-]+)\/([a-z-]+)$/', $composer['name'], $m)) {
+    fwrite(
+        STDERR,
+        'Cannot extract repository name from composer.json, please check value of "name" key in that file.'
+    );
+    exit(1);
+}
+$org = $m[1];
+$repo = $m[2];
+
+// if there is no docs directory create one
+if (! is_dir('docs')) {
+    // if there is doc directory rename it to docs
+    if (is_dir('doc')) {
+        rename('doc', 'docs');
+    } else {
+        mkdir('docs', 0775);
+    }
+}
+
+$docs = [
+    'CODE_OF_CONDUCT.md',
+    'CONTRIBUTING.md',
+    'ISSUE_TEMPLATE.md',
+    'PULL_REQUEST_TEMPLATE.md',
+    'SUPPORT.md',
+];
+
+$replace = [
+    '{org}' => $org,
+    '{repo}' => $repo,
+    '{category}' => strpos($repo, 'zend-expressive') === 0 ? 'expressive' : 'components',
+];
+
+foreach ($docs as $file) {
+    if (file_exists('docs/' . $file)) {
+        unlink('docs/' . $file);
+    }
+
+    $content = file_get_contents(__DIR__ . '/../template/docs/' . $file);
+    $content = strtr($content, $replace);
+
+    file_put_contents('docs/' . $file, $content);
+}
+
+if (file_exists('CONTRIBUTING.md')) {
+    unlink('CONTRIBUTING.md');
+}
+
+if (file_exists('CONDUCT.md')) {
+    unlink('CONDUCT.md');
+}
+
+// Update LICENSE.md
+
+$currentYear = date('Y');
+$year = $currentYear;
+if (file_exists('LICENSE.md')) {
+    $content = file_get_contents('LICENSE.md');
+    if (preg_match('/Copyright \(c\) (\d{4})/', $content, $m)) {
+        $year = $m[1];
+    } else {
+        fwrite(STDERR, 'Cannot match year in current LICENSE.md file. Using current year only.' . PHP_EOL);
+    }
+}
+
+$content = file_get_contents(__DIR__ . '/../template/LICENSE.md');
+$content = str_replace(
+    '{year}',
+    $year < $currentYear ? $year . '-' . $currentYear : $currentYear,
+    $content
+);
+
+file_put_contents('LICENSE.md', $content);
+
+// .coveralls.yml
+file_put_contents('.coveralls.yml', file_get_contents(__DIR__ . '/../template/.coveralls.yml'));
+
+// check if repository has documentation
+$hasDocs = file_exists('mkdocs.yml');
+
+// .gitattributes
+$content = file_get_contents(__DIR__ . '/../template/.gitattributes');
+if (! $hasDocs) {
+    $content = preg_replace('/\/mkdocs\.yml export-ignore\s+/', '', $content);
+}
+if (file_exists('phpbench.json')) {
+    if (file_exists('benchmark') && is_dir('benchmark')) {
+        $content = '/benchmark export-ignore' . PHP_EOL . $content;
+    }
+    if (file_exists('benchmarks') && is_dir('benchmarks')) {
+        $content = '/benchmarks export-ignore' . PHP_EOL . $content;
+    }
+    $content = preg_replace(
+        '/\/phpcs\.xml export-ignore/',
+        '/phpbench.json export-ignore' . PHP_EOL . '/phpcs.xml export-ignore',
+        $content
+    );
```

Comment:
It will be then alphabetically but files will be mixed with dirs. Usually files are sorted - first directories, then files (this is what I can see in project struct in PHPStorm.

I'll check what I can do with it, I agree that it's not so clear right now.

Reply:
Directories will have a `/` following them generally, which should make sorting work.